runtime/lua/vim/lsp/completion.lua

```diff
@@ -244,6 +244,44 @@ local function match_item_by_value(value, prefix)
   return vim.startswith(value, prefix)
 end
 
+--- Generate the highlight group name for LSP completion item kind.
+--- Checks for the existence of specific highlight groups
+--- and returns the appropriate kind and kind_hlgroup
+---
+--- @param item lsp.CompletionItem
+--- @return string|nil, string|nil
+local function generate_kind(item)
+  local kind = lsp.protocol.CompletionItemKind[item.kind]
+  if not kind then
+    return
+  end
+
+  if item.kind == lsp.protocol.CompletionItemKind.Color then
+    local doc = get_doc(item)
+    if #doc > 0 then
+      local r, g, b = doc:match('rgb%((%d+)%s*,?%s*(%d+)%s*,?%s*(%d+)%)')
+      if r then
+        local color = string.format('%02x%02x%02x', tonumber(r), tonumber(g), tonumber(b))
+        local group = ('@lsp.color.%s'):format(color)
+        if vim.fn.hlID(group) < 1 then
+          api.nvim_set_hl(0, group, { fg = '#' .. color })
+        end
+        return '■', group
+      end
+    end
```

Comment:
What kind of server is providing the color codes in the document property?

I just tested this with the css server because I was curios how this looks like, but there the color codes are in the text:

![image](https://github.com/user-attachments/assets/06810116-0cb3-4158-a189-67312391877a)

No strong opinion on whether we should do this at all.

Reply:
from tailwindcss..seems like also can do some work on css. I remember vscode do it

Reply:
![Uploading cfef1f1ccd0a1fd6569419faaada7ab8.png…]()